===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ PortObjectAddRule:n$3 == { -1 } }) free(*(PortObjectAddRule:pruleid)) at 3 (line 1747, column 5) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
+++ b/table1/benchmarks/snort-2.9.13/src/sfutil/sfportobject.c
@@ -1742,7 +1742,8 @@ int PortObjectAddRule( PortObject * po , int rule )
 
     *pruleid = rule;
 
-    sflist_add_tail( po->rule_list, pruleid );
+    if (sflist_add_tail( po->rule_list, pruleid ) == -1)
+        free(pruleid);
 
     return 0;
 }
